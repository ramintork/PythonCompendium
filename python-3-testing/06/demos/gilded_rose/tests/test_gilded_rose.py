# -*- coding: utf-8 -*-
import unittest

import approvaltests

from gilded_rose import Item, GildedRose


def print_item(item):
    return f"Item({item.name}, sell_in={item.sell_in}, quality={item.quality})"


def update_quality_printer(args, result):
    return f"{args} => {print_item(result)}\n"


def test_update_quality():
    names = ["foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"]
    sell_ins = [-1, 0, 1, 6, 11]
    qualitys = [0, 1, 2, 48, 49, 50]

    approvaltests.verify_best_covering_pairs(
        update_quality_for_item,
        [
            names,
            sell_ins,
            qualitys
        ],
        formatter=update_quality_printer
    )


def update_quality_for_item(name, sell_in, quality):
    item = Item(name, sell_in, quality)
    items = [item]
    gilded_rose = GildedRose(items)
    gilded_rose.update_quality()
    return item
