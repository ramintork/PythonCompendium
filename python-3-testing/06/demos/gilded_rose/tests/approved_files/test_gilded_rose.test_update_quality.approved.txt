Testing an optimized 30/120 scenarios:

['foo', -1, 0] => Item(foo, sell_in=-2, quality=0)
['Aged Brie', 0, 0] => Item(Aged Brie, sell_in=-1, quality=2)
['Backstage passes to a TAFKAL80ETC concert', 1, 0] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=0, quality=3)
['Sulfuras, Hand of Ragnaros', 6, 0] => Item(Sulfuras, Hand of Ragnaros, sell_in=6, quality=0)
['Sulfuras, Hand of Ragnaros', 11, 1] => Item(Sulfuras, Hand of Ragnaros, sell_in=11, quality=1)
['Backstage passes to a TAFKAL80ETC concert', 11, 2] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=10, quality=3)
['Aged Brie', -1, 2] => Item(Aged Brie, sell_in=-2, quality=4)
['foo', 6, 2] => Item(foo, sell_in=5, quality=1)
['foo', 1, 1] => Item(foo, sell_in=0, quality=0)
['Aged Brie', 1, 48] => Item(Aged Brie, sell_in=0, quality=49)
['Backstage passes to a TAFKAL80ETC concert', 0, 1] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=-1, quality=0)
['Sulfuras, Hand of Ragnaros', 0, 2] => Item(Sulfuras, Hand of Ragnaros, sell_in=0, quality=2)
['Sulfuras, Hand of Ragnaros', -1, 48] => Item(Sulfuras, Hand of Ragnaros, sell_in=-1, quality=48)
['Backstage passes to a TAFKAL80ETC concert', 6, 48] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=5, quality=50)
['Aged Brie', 11, 49] => Item(Aged Brie, sell_in=10, quality=50)
['foo', 11, 48] => Item(foo, sell_in=10, quality=47)
['foo', 0, 49] => Item(foo, sell_in=-1, quality=47)
['Aged Brie', 6, 1] => Item(Aged Brie, sell_in=5, quality=2)
['Backstage passes to a TAFKAL80ETC concert', -1, 49] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=-2, quality=0)
['Sulfuras, Hand of Ragnaros', 1, 49] => Item(Sulfuras, Hand of Ragnaros, sell_in=1, quality=49)
['Sulfuras, Hand of Ragnaros', 1, 50] => Item(Sulfuras, Hand of Ragnaros, sell_in=1, quality=50)
['Backstage passes to a TAFKAL80ETC concert', -1, 50] => Item(Backstage passes to a TAFKAL80ETC concert, sell_in=-2, quality=0)
['Aged Brie', 6, 50] => Item(Aged Brie, sell_in=5, quality=50)
['foo', 0, 50] => Item(foo, sell_in=-1, quality=48)
['foo', 11, 50] => Item(foo, sell_in=10, quality=49)
['foo', 11, 0] => Item(foo, sell_in=10, quality=0)
['foo', 0, 48] => Item(foo, sell_in=-1, quality=46)
['foo', 6, 49] => Item(foo, sell_in=5, quality=48)
['foo', -1, 1] => Item(foo, sell_in=-2, quality=0)
['foo', 1, 2] => Item(foo, sell_in=0, quality=1)
